src/client/client.ts

```diff
@@ -29,20 +29,22 @@ export type ClientOptions = {
    */
   triggerHydration?: TriggerHydration
   ISLAND_FILES?: Record<string, () => Promise<unknown>>
+  /**
+   * @deprecated
+   */
   island_root?: string
 }
 
 export const createClient = async (options?: ClientOptions) => {
   const FILES = options?.ISLAND_FILES ?? {
-    ...import.meta.glob('/app/islands/**/[a-zA-Z0-9[-]+.(tsx|ts)'),
-    ...import.meta.glob('/app/routes/**/_[a-zA-Z0-9[-]+.island.(tsx|ts)'),
+    ...import.meta.glob('/app/islands/**/[a-zA-Z0-9-]+.(tsx|ts)'),
+    ...import.meta.glob('/app/**/_[a-zA-Z0-9-]+.island.(tsx|ts)'),
+    ...import.meta.glob('/app/**/$[a-zA-Z0-9-]+.(tsx|ts)'),
```

Comment:
It has nothing to do with this PR change, In `matchIslandComponentId()`, only `.tsx` matches (`.ts does not), so I think only `.tsx` should be used here as well.

https://github.com/honojs/honox/pull/176/files#diff-4b7b8acce4bb320abe3a28d1bb85d7ad346253af647225df7e62d19891192a27R25

Reply:
@usualoma 

You are right! It should be only `.tsx` and it's not a problem if it does not support `.ts`.